src/CMakeLists.txt

```diff
@@ -1,107 +1,85 @@
 # Must be included before CMAKE_INSTALL_INCLUDEDIR is used.
 include(GNUInstallDirs)
-set(${PROJECT_NAME}_installables "")
 
 if(SECP256K1_ASM STREQUAL "arm")
-  add_library(common OBJECT
+  target_sources(secp_common_objs PRIVATE
     asm/field_10x26_arm.s
   )
-  set(common_obj "$<TARGET_OBJECTS:common>")
-else()
-  set(common_obj "")
 endif()
 
-add_library(precomputed OBJECT
+add_library(secp_precomputed_objs OBJECT EXCLUDE_FROM_ALL
   precomputed_ecmult.c
   precomputed_ecmult_gen.c
 )
-set(internal_obj "$<TARGET_OBJECTS:precomputed>" "${common_obj}")
 
-add_library(secp256k1 SHARED EXCLUDE_FROM_ALL
-  secp256k1.c
-  ${internal_obj}
+add_library(secp_common_objs OBJECT EXCLUDE_FROM_ALL secp256k1.c)
+
+add_library(secp_interface INTERFACE)
+target_link_libraries(secp_common_objs PRIVATE secp_interface)
+target_link_libraries(secp_precomputed_objs PRIVATE secp_interface)
+
+# Add objects explicitly rather than linking to the object libs to keep them
+# from being exported.
+add_library(secp256k1 $<TARGET_OBJECTS:secp_precomputed_objs> $<TARGET_OBJECTS:secp_common_objs>)
+
+target_compile_definitions(secp_interface INTERFACE
+  $<$<C_COMPILER_ID:MSVC>:_CRT_SECURE_NO_WARNINGS>
```

Comment:
@hebasto What is this btw? I don't see it in the autotools build and I suspect I didn't handle it correctly here.

Reply:
During working on #1113, at some point in the past, I had a ["unsafe CRT Library function"](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4996?view=msvc-170#unsafe-crt-library-functions) warning. However, I'm not able to reproduce it now.

Reply:
FWIW, a similar suppression is used in [Bitcoin Core](https://github.com/bitcoin/bitcoin/blob/c7f1d95f52883d7087b74f45f5e4ce1100d51149/build_msvc/common.init.vcxproj.in#L93):
```XML
      <PreprocessorDefinitions>_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING;ZMQ_STATIC;NOMINMAX;WIN32;HAVE_CONFIG_H;_CRT_SECURE_NO_WARNINGS;_SCL_SECURE_NO_WARNINGS;_CONSOLE;_WIN32_WINNT=0x0601;_WIN32_IE=0x0501;WIN32_LEAN_AND_MEAN;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Reply:
I realize this is kinda unrelated here, but it snuck in in #1113 so we may as well address it.

Due to the nature of that suppression, I think we'd rather know about it and address it. I'm tempted to remove this as part of this PR and ask you to submit a new one to rea-add it if you're ever able to reproduce again. That way we can debate the merit/alternatives.

Thoughts on that?

Reply:
> I'm tempted to remove this as part of this PR...

Agree.